central/notifier/datastore/datastore_impl.go

```diff
@@ -22,6 +24,29 @@ type datastoreImpl struct {
 	storage store.Store
 }
 
+func verifyOrigin(ctx context.Context, n *storage.Notifier) error {
+	if !declarativeconfig.CanModifyResource(ctx, n) {
+		return errox.NotAuthorized.Newf("notifier %q's origin is %s, "+
+			"cannot be modified or deleted with the current permission",
+			n.GetName(), n.GetTraits().GetOrigin())
+	}
+	return nil
+}
+
+func (b *datastoreImpl) verifyExistsAndMutable(ctx context.Context, id string) (*storage.Notifier, error) {
+	notifier, exists, err := b.storage.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errox.NotFound.Newf("notifier with id %q was not found", id)
+	}
+	if notifier.GetTraits().GetMutabilityMode() != storage.Traits_ALLOW_MUTATE {
```

Comment:
`Traits.MutabilityMode` was our first solution when we needed to introduce objects that shouldn't be deleted via UI. We needed that for ACSCS auth provider, and now this auth provider will be moved to be declarative instead. I suggest we create ticket to remove all usages of mutability mode once we'll move away from using it for ACSCS.

In current context, I don't think `mutabilityMode` flag should be supported in API. Let's just ignore it, or even throw an error if someone tries to set it to non-default value.